src/diff_edit.rs

```diff
@@ -96,6 +135,122 @@ fn set_readonly_recursively(path: &Path) -> Result<(), std::io::Error> {
+        _ => {
+            // In this case, `materialized_conflict` is a short summary of what kind of
+            // files are involved in the conflict.
+            let summary = match std::str::from_utf8(materialized_conflict.as_slice()) {
+                Ok(s) if !materialized_conflict.is_empty() => s.to_string(),
+                _ => "<Failed to create a readable summary of the conflict>".to_string(),
+            };
+            return Err(DiffEditError::NotNormalFilesError(
+                repo_path.clone(),
+                summary,
+            ));
+        }
+    };
+    if content.removes.len() > 1 || content.adds.len() > 2 {
+        return Err(DiffEditError::ConflictTooComplicatedError {
+            path: repo_path.clone(),
+            removes: content.removes.len(),
+            adds: content.adds.len(),
+        });
+    };
+
+    let temp_dir = tempfile::Builder::new()
+        .prefix("jj-merge-")
+        .tempdir()
+        .map_err(DiffEditError::SetUpDirError)?;
+    let files_maybe = [
+        ("base", content.removes.get(0)),
+        ("left", content.adds.get(0)),
+        ("right", content.adds.get(1)),
+        ("output", Some(&materialized_conflict)),
+    ];
+    let files: HashMap<&str, _> = files_maybe
+        .map(|(name, f)| (name, f.cloned().unwrap_or_default()))
+        .iter()
+        .cloned()
+        .collect();
+    let paths: Result<HashMap<&str, _>, DiffEditError> = files
+        .iter()
+        .map(|(name, contents)| {
+            let path = temp_dir.path().join(name);
+            let mut file = File::create(&path).map_err(DiffEditError::SetUpDirError)?;
+            file.write_all(contents)
+                .map_err(DiffEditError::SetUpDirError)?;
+            file.sync_all()?;
+            drop(file);
+            if *name != "output" {
+                // TODO: Should actually ignore the error here, or have a warning.
+                set_readonly_recursively(&path)?;
+            }
+            Ok((*name, path))
+        })
+        .collect();
+    let paths = paths?;
+
+    let progname = "vimdiff";
+    let exit_status = Command::new(progname)
+        .args(["-f", "-d"])
+        .arg(paths.get("output").unwrap())
+        .arg("-M")
+        .args(["left", "base", "right"].map(|n| paths.get(n).unwrap()))
+        .args(["-c", "wincmd J", "-c", "setl modifiable write"])
+        .status()
+        .map_err(|e| DiffEditError::ExecuteEditorError {
+            editor_binary: progname.to_string(),
+            source: e,
+        })?;
+    if !exit_status.success() {
+        return Err(DiffEditError::DifftoolAborted);
+    }
+
+    let mut output_file_contents: Vec<u8> = vec![];
+    _ = File::open(paths.get("output").unwrap())?.read_to_end(&mut output_file_contents)?;
+    if output_file_contents.is_empty() || output_file_contents == materialized_conflict {
+        return Err(DiffEditError::EmptyOrUnchanged);
+    }
+    // TODO: parse any remaining conflicts
```

Comment:
You should be able to use a regular editor as a merge tool already with this PR with an appropriate (possibly non-obvious) config.

```toml
[ui]
merge-tool="vim"

[merge-tools]
vim.merge-args = ["$output"]
vim.merge-tool-edits-conflict-markers=true
```

Reply:
I'm not sure this is too magical, but we could default to this kind of config if `tool.merge-args` is empty. The other option is to error out in this case.

The main problem is that this would mean that `tool.merge-tool-edits-conflict-markers` would default to `false` if `tool.merge-args` is specified, but it would default to `true` if the args are not specified.

Reply:
A better option might be to have a special tool that can be used with `merge-tool="jj-editor"` that has

```toml
[merge-tools]
# jj-editor.program is set to be the same editor jj uses
jj-editor.merge-args = ["$output"]
jj-editor.merge-tool-edits-conflict-markers=true
```

I might add that to this PR.

Reply:
Actually, I think this should be a command-line option, e.g. `jj resolve --editor`, since we probably want this to work even when resolving conflicts with more than 2 adds. The configuration approach would need even more special-casing.

I'll put in a TODO and mark this resolved.

Reply:
> Actually, I think this should be a command-line option, e.g. `jj resolve --editor`

I had the same thought, so that sounds good to me (as a TODO, as you said).